# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1276211381 14400
#      Thu Jun 10 19:09:41 2010 -0400
# Node ID 701b969db6a3eb7e8724b98878e3583c3e0a170a
# Parent  d97165928d052b65b0b0f0dab9f8ca28b23ce492
data.cc: Fix test for concatentating empty nd-arrays.

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2010-06-10  Ben Abbott <bpabbott@mac.com>
+
+	* data.cc: Fix test for concatentating empty nd-arrays.
+
 2010-06-10  Jaroslav Hajek  <highegg@gmail.com>
 
 	* data.cc (Fvec): New DEFUN.
 
 2010-06-09  Jaroslav Hajek  <highegg@gmail.com>
 
 	* data.cc (Freshape): Use octave_idx_type rather than int.
 	Don't use intermediate Array when sizes are given as separate
diff --git a/src/data.cc b/src/data.cc
--- a/src/data.cc
+++ b/src/data.cc
@@ -1807,17 +1807,17 @@ cat (4, ones(2, 2), zeros (2, 2))\n\
 %!assert (testcat('uint64', 'double', 'uint64', false));
 %!assert (testcat('single', 'uint64', 'uint64', false));
 %!assert (testcat('uint64', 'single', 'uint64', false));
 %!assert (testcat('uint64', 'uint64', 'uint64', false));
 
 %! assert (cat (3, [], [1,2;3,4]), [1,2;3,4]);
 %! assert (cat (3, [1,2;3,4], []), [1,2;3,4]);
 %! assert (cat (3, [], [1,2;3,4], []), [1,2;3,4]);
-%! assert (cat (3, [], [], []), []);
+%! assert (cat (3, [], [], []), zeros (0, 0, 3));
 
 */
 
 static octave_value
 do_permute (const octave_value_list& args, bool inv)
 {
   octave_value retval;
 
